### ISSUE TITLE
Adding HCA as datasource to Zooma

### ISSUE THREAD

**Issue by rays22** [ebi-ait/hca-ebi-wrangler-central#104]
**Description of the task:**

1. Use the attached csv template for loading annotations into Zooma.

2. You can find a description of the values that need be supplied in the csv file at https://www.ebi.ac.uk/spot/zooma/docs/getting-started.html under the section "Adding New Mappings to Zooma".

3. Once you have a populated csv file, tell the SPOT team the URL from where they can load this file from.
This URL needs to be constant and cannot change with different releases from HCA. Many of Zooma datasources are loaded from `ftp://ftp.ebi.ac.uk/pub/databases/spot/zooma/data....`
So for example `ftp://ftp.ebi.ac.uk/pub/databases/spot/zooma/data/annotations/hca/latest/hca_annotations.csv` may be viable option.

4. Once we have provided the SPOT team with the information in (3), they will load it into their test environment to give us an opportunity to review and spot any possible mistakes.

5. Once you give the SPOT team the go-ahead on 4, they will make it available on prod and notify us.

I think we need to plan the update-upload-testing cycle as a recurring process. 

**Acceptance criteria for the task:**

- [ ] complete the first upload, testing and production cycle
- [ ] establish process for update-upload-testing cycles

**Comment by rays22** [comment]
`zooma-template.csv`:

```
ANNOTATION_URI,ANNOTATION_ID,STUDY_URI,STUDY,STUDY_ID,STUDY_TYPE,BIOENTITY_URI,BIOENTITY,BIOENTITY_TYPE_URI,BIOENTITY_TYPE_NAME,BIOENTITY_ID,PROPERTY_URI,PROPERTY_ID,PROPERTY_TYPE,PROPERTY_VALUE,SEMANTIC_TAG,ANNOTATOR,ANNOTATION_DATE
```

**Comment by lauraclarke** [comment]
@rays22 can you add details here of what you need @aaclan-ebi's help with

**Comment by rays22** [comment]
I need help on how I can get all the ontology related metadata field values for all the datasets that we have wrangled in bulk.

```
*.text *.ontology *.ontology_label
```


I have tested it and I can get that information for each dataset by using the Ingest API if I know the `submissionEnvelope` and query the appropriate endpoints:

    "submissionEnvelope" : {
      "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX",
      "title" : "A single submission envelope"
    },
    "biomaterials" : {
      "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/biomaterials"
    },
    "processes" : {
      "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/processes"
    },
    "files" : {
      "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/files",
      "title" : "Access or create, within a submission envelope, a new assay"
    },
    "projects" : {
      "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/projects",
      "title" : "Access or create projects. Creation can only be done inside a submission envelope"
    },
    "protocols" : {
      "href" : "https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/XXXXXXXXXXXXXXXXXXXXXXXX/protocols",
      "title" : "Access or create protocols"


The problem is that I have not found a trivial way to get a list of all the `submissionEnvelope` values for all the wrangled datasets. If I go to Ingest UI and log in, then I can open each project one-by-one and then I can find the `submissionEnvelope` for the latest submission for that project.
I am not sure how long it would take for me to find all the required `submissionEnvelope` IDs for all the wrangled datasets this way.

I would like to ask @aaclan-ebi  if there is a better and faster way to get the `submissionEnvelope` IDs for all the wrangled projects/latest submissions.

Alternatively, there may be completely different and better ways to get the data in bulk from the ontology related fields, e.g Mongo DB? Or Terra Data Repo?
Actually I could write more advanced SQL queries more quickly than scripts to parse JSON responses...  

Shall I make another ticket about getting the bulk data? If yes, would it be a development ticket or operations?

**Comment by mshadbolt** [comment]
I got a list of submission envelope IDs and am using this script to get all the relevant ontology mappings : https://github.com/HumanCellAtlas/hca-data-wrangling/blob/master/src/ontology_mappings_extractor.py

@rays22 are you aware of whether the csv that we provide should be non-duplicated, i.e. each match between a provided text field and an ontology occurs only once, or should appear as many times as in our data so that it provides more weight to particular annotations?

**Comment by rays22** [comment]
Thanks @mshadbolt .
I think the  `text label -- preferred ontology term` pairs should be unique in the csv.
As I understand it, a particular mapping will get more weight in zooma if it appears in multiple independent mappings, not when it appears in multiple times in one source.
I think we will be able to select the HCA-specific mappings, so those HCA-preferred mappings should appear as top hits.

**Comment by clairerye** [comment]
Created new ticket in product development https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/230
